### PR TITLE
changes for langauge translation

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

File: molo/core/middleware.py [hunk 1]
```diff
@@ -150,6 +150,11 @@ def process_response(self, request, response):
         return response
 
 
+class WillReturnTrue(django.utils.deprecation.MiddlewareMixin):
```
Comment (sewagodimo) [inline molo/core/middleware.py]: ooh

File: molo/core/models.py [hunk 2]
```diff
@@ -556,13 +556,57 @@ class LanguageRelation(models.Model):
...
         return "get_translation_for_{}_{}_{}_{}_{}".format(
             self.pk, locale, site.pk, is_live,
             self.latest_revision_created_at.isoformat())
 
     def get_translation_for(self, locale, site, is_live=True):
+        if settings.USE_NEW_TRANSLATIONS:
+            return self.get_translation_for_new(locale, site, is_live)
+        return self.get_translation_for_old(locale, site, is_live)
+
+    def get_translation_for_new(self, locale, site, is_live=True):
```
Comment (sewagodimo) [inline molo/core/models.py]: should the default for is_live not be False?

File: molo/core/templatetags/core_tags.py [hunk 3]
```diff
@@ -36,12 +40,17 @@ def get_pages(context, queryset, locale):
     show_only_translated_pages = SiteSettings.for_site(
         request.site).show_only_translated_pages
     pages = []
-    for page in queryset:
-        translation = page.get_translation_for(locale, request.site)
-        if translation and translation.live:
-            pages.append(translation)
-        elif page.live and not show_only_translated_pages:
-            pages.append(page)
+    if settings.USE_NEW_TRANSLATIONS:
```
Comment (sewagodimo) [inline molo/core/templatetags/core_tags.py]: there should be an else for` if settings.USE_NEW_TRANSLATIONS:`

File: molo/core/views.py [hunk 4]
```diff
@@ -136,8 +136,10 @@ def add_translation(request, page_id, locale):
     language_relation = translation.languages.first()
     language_relation.language = language
     language_relation.save()
+    page.translated_pages.add(translation)
```
Comment (sewagodimo) [inline molo/core/views.py]: So the templates should also be updated to relfect that the translations are done in the view

File: molo/core/models.py [hunk 5]
```diff
@@ -1370,8 +1420,9 @@ class ArticlePageMetaDataTag(TaggedItemBase):
 
 class ArticlePage(ImportableMixin, CommentedPageMixin,
                   TranslatablePageMixin, PageEffectiveImageMixin, MoloPage):
+    language = models.ForeignKey('core.SiteLanguage', blank=True, null=True)
```
Comment (sewagodimo) [inline molo/core/models.py]: fields to add

File: local_test_settings.py [hunk 6]
```diff
@@ -7,7 +7,7 @@
     'site2',
 ]
 
-ENABLE_SSO = True
+ENABLE_SSO = False
```
[the remark this answers is not in the document]
Comment (KaitCrawford) [reply local_test_settings.py]: Single sign on. I'd guess that this setting is what controls the redirect to mission control when an admin user tried to login.

[the remark this answers is not in the document]
Comment (sewagodimo) [reply local_test_settings.py]: ooh, should this fix not be in another PR then?

File: molo/core/models.py (hunk 2 above)
Comment (KaitCrawford) [reply molo/core/models.py]: No. by default we only want to get the live pages.

File: molo/core/templatetags/core_tags.py (hunk 3 above)
Comment (KaitCrawford) [reply molo/core/templatetags/core_tags.py]: there is? it's on line 47

File: molo/core/models.py (hunk 2 above)
Comment (sewagodimo) [reply molo/core/models.py]: I thought is live meant like whether you are in production

Comment (KaitCrawford) [reply molo/core/models.py]: nope. it's a flag to show if we only want the translated pages that are live, ie they have been published

File: molo/core/models.py [hunk 7]
```diff
@@ -556,13 +556,57 @@ class LanguageRelation(models.Model):
     language = models.ForeignKey('core.SiteLanguage', related_name='+')
 
 
+def get_translation_for_new_qs(qs, locale, site, is_live=True):
```
Comment (sewagodimo) [inline molo/core/models.py]: should I add this to my PR @KaitCrawford for molo.core translation fields.

Comment (KaitCrawford) [reply molo/core/models.py]: @sewagodimo no. this part of the code is for the ticket I'm working on. Your ticket is for adding the fields to all the models that inherit from TranslatablePageMixin

**Pull request closed without merging**